### PR TITLE
[BD-27] [TNL-7948] Fix Static File References

### PR DESCRIPTION
**Pull request by aayushagra**
We noticed that not all static links were escaped as part of the conversion process. This PR adds code to recursively process all input going into OLX nodes to automatically escape any static file references.

**JIRA tickets**: TNL-7948

**Merge deadline**: None

**Testing instructions**:

1. Download and try to import a course with static links in various places, including in multiple-choice problems answers (Where we found the problem), discussion topics & other locations
2. Convert the course with cc2olx and verify that all static links were escaped successfully

**Reviewers**
- [x] @farhaanbukhsh

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @aayushagra! I've created [BLENDED-772](https://openedx.atlassian.net/browse/BLENDED-772) to keep track of it in Jira.   More details are on the [BD-27](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1544912903/BD-27+Common+Cartridge+Converter) project page.


When this pull request is ready, tag your edX technical lead.

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
I have added few nits, I tested this and it looks good with the course in question. Good work :medal_military:

**Comment by farhaanbukhsh** [comment]
@aayushagra  can you resolve the conflicts

File: src/cc2olx/olx.py [hunk 1]
```diff
@@ -155,17 +155,38 @@ def process_ims_cc_filebase(item, html):
 
         return html
 
+    def _process_static_links_from_details(self, details):
+        """
+        Take a variable and recursively find & escape all static links within strings
+
+        Input: {'problem_details': '<img src="$IMS-CC-FILEBASE$/..."/> Lorem Ipsum', ...}
+        Output: {'problem_details': '<img src="/static/..."/> Lorem Ipsum', ...}
```
Comment (farhaanbukhsh) [inline src/cc2olx/olx.py:163]: There is a specific format in which you should write the docstring. Have a look at https://github.com/edx/cc2olx/blob/master/src/cc2olx/filesystem.py#L19

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
Some more nits :) please have a look.

File: src/cc2olx/olx.py [hunk 2]
```diff
@@ -162,6 +162,31 @@ def process_ims_cc_filebase(item, html):
 
         return html
 
+    def _process_static_links_from_details(self, details):
+        """
+        Take a variable and recursively find & escape all static links within strings
+
+        Args:
+            self: self
+            details: A dictionary or an array of dictionaries containing node data
```
Comment (farhaanbukhsh) [inline src/cc2olx/olx.py:171]:
```suggestion
            details: A dictionary or list of dictionaries containing node data.
```

File: src/cc2olx/olx.py [hunk 3]
```diff
@@ -162,6 +162,31 @@ def process_ims_cc_filebase(item, html):
...
+
+        Returns:
+            details: Returns the same details data but every string with a static link
+                        inside it has been escaped to an OLX friendly format
```
Comment (farhaanbukhsh) [inline src/cc2olx/olx.py:175]:
```suggestion
        Returns:
            details: Returns detail data with static link
                        escaped to an OLX-friendly format.
```

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- [x] I tested this with the course in question, able to import it properly
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [ ] ~~Includes documentation~~
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

**Review by alangsto — APPROVED** [review]

**Pull request merged**